### PR TITLE
feat: ENT-6843 dispatched event for subsidy redemption

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
Description: This PR adds an event on the openedx event bus upon redemption of a subsidy

JIRA: https://2u-internal.atlassian.net/browse/ENT-6843

### CODE REVIEW

File: enterprise_access/apps/api/v1/views.py [hunk 1]
```diff
@@ -826,7 +828,11 @@ def redeem(self, request, *args, **kwargs):
                 )
             if policy.can_redeem(learner_id, content_key):
                 response = policy.redeem(learner_id, content_key)
-
+                if settings.KAFKA_ENABLED:  # pragma: no cover
```
Comment (muhammad-ammar) [inline enterprise_access/apps/api/v1/views.py:831]: I think we should move the `if` condition inside the `send_subsidy_redemption_event_to_event_bus`. The caller of this method should not take the responsibility to check if `Kafka` is enabled/disabled.

**Review by muhammad-ammar — APPROVED** [review]
LGTM. Left one minor comment.

**Review by saleem-latif — APPROVED** [review]

File: enterprise_access/apps/api/v1/views.py (hunk 1 above)
Comment (irfanuddinahmad) [reply enterprise_access/apps/api/v1/views.py]: good point ... done

File: enterprise_access/apps/api/serializers.py [hunk 2]
```diff
@@ -123,7 +123,7 @@ def update(self, instance, validated_data):
         return super().update(instance, validated_data)
 
 
-class SubsidiyAccessPolicyRedeemSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+class SubsidyAccessPolicyRedeemSerializer(serializers.Serializer):  # pylint: disable=abstract-method
```
Comment (iloveagent57) [inline enterprise_access/apps/api/serializers.py:126]: ah, good catch

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**